src/main/kotlin/no/nav/helse/spade/behandlinger/KafkaBehandlingerRepository.kt

```diff
@@ -32,19 +41,12 @@ class KafkaBehandlingerRepository(stream: BehandlingerStream) {
    }
 
    fun getBehandlingerForPeriode(fom: String, tom: String): Either<Feilårsak, List<JsonNode>> = try {
-      stateStore.all().asSequence().filter { keyval ->
-         keyval.value.any {
-            if (it.has("avklarteVerdier") && it.path("avklarteVerdier").has("medlemsskap")
-               && it.path("avklarteVerdier").path("medlemsskap").has("vurderingstidspunkt")) {
-               val vurderingstidspunkt = it.path("avklarteVerdier").path("medlemsskap").get("vurderingstidspunkt").textValue()
-               val vurderingstidspunktWithoutTimestamp = vurderingstidspunkt.substring(0, 10)
-               isDateInPeriod(vurderingstidspunktWithoutTimestamp, fom, tom)
-            } else {
-               false
-            }
+      val initialList: MutableList<JsonNode> = mutableListOf()
+      stateStore.all().asSequence().fold(initialList) { acc, keyval ->
+         keyval.value.forEach { node ->
+            if (getVurderingstidspunkt(node)?.let { isDateInPeriod(it, fom, tom) } == true) acc.add(node)
```

Comment:
Jeg lurer på om det hadde blitt penere kode av å filtrere etter å ha flatMap-et i stedet for å fold-e over topnivået? I hvert fall, API-et til fold og lignende funksjoner er vel at man returnerer ny akkumulert verdi, ikke at man opererer på akkumulatoren. Så her burde det ha vært filtrering i stedet for forEach, og returnering av acc + outcome av filtreringen. Men uansett blir det litt kronglete kode, jeg tror det hadde hjulpet å gjøre datosjekken på riktig nivå.

Reply:
Å filtrere listen to ganger altså? En gang før og etter flatMap? Var det jeg prøvde å unngå. Ellers enig i at jeg kan returnere acc + outcome av filtreringen

Reply:
Nei, jeg mente én filtrering. Men på generelt grunnlag mener jeg at lesbarhet trumfer lavt antall cpu-sykler :)